syntax="proto3";

service RepoService {
   
	rpc Save(User) returns (User) {}
	rpc FindById(User_Id) returns (User) {}
	rpc FindByEmail(UserEmail) returns (User) {}
	rpc ResetDatabase(Empty) returns (Empty) {}

}

message User {
	uint32 id = 1;
	string name = 2;
	string email = 3;
	string password = 4;
}

message User_Id {
	uint32 user_id = 1;
}

message UserEmail {
	string user_email = 1;
}

message Empty {}